internal/pkg/plan/persist.go

```diff
@@ -53,18 +51,54 @@ func (b *persistPlanBuilder) build() {
 			b.actions = append(b.actions, &DeleteRecordAction{Record: record})
 		}
 	}
+
+	// Sort actions by action order AND path
+	sort.SliceStable(b.actions, func(i, j int) bool {
+		iAction := b.actions[i]
+		jAction := b.actions[j]
+		orderDiff := iAction.Order() - jAction.Order()
+		if orderDiff != 0 {
+			return orderDiff < 0
+		}
+		return iAction.Path() < jAction.Path()
+	})
 }
 
-func (b *persistPlanBuilder) tryAddConfig(projectPath string, branch *model.BranchState) []PersistAction {
-	// Is path from the branch dir?
-	parentPath := branch.Path()
-	relPath, err := filesystem.Rel(parentPath, projectPath)
+func (b *persistPlanBuilder) tryAdd(fullPath string, parent model.RecordPaths) {
+	// Is path from the parent?
+	parentPath := parent.Path()
+	if !filesystem.IsFrom(fullPath, parentPath) {
+		return
+	}
+	objectPath, err := filesystem.Rel(parentPath, fullPath)
 	if err != nil {
 		b.errors.Append(err)
-		return nil
+		return
+	}
+	path := model.NewPathInProject(parent.Path(), objectPath)
+
+	// Add object according to the parent type
+	switch p := parent.(type) {
+	case *model.BranchState:
```

Comment:
Na zaklade typu `parent` objektu, sa hladaju pod-objekty.

Reply:
Ziadne vnorene `for` slucky, je to takto prehladnejsie ....  a hlavne jednoduchsie tu viem pridat hladanie `variables`.